apache/superset
# fix: ensure numeric values for extra metadata_cache_timeout payloads

### SUMMARY
We identified an issue where FileSystemCache is used, setting values for `table_cache_timeout` and `schema_cache_timeout` in the database properties modal leads to failures in fetching schema and table lists in SQLLab. The requests to `/api/v1/database/{dbId}/tables/?q=(force:!f,schema_name:{schemaName})` were returning a 422 UNPROCESSABLE ENTITY status.

The root cause of the issue was that `table_cache_timeout` and `schema_cache_timeout` values were being serialized as strings instead of numbers. The backend expected numeric values, resulting in the 422 error when it encountered strings. To resolve this, we need to ensure that these values are correctly converted to numbers before they are sent in the payload.


### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before

https://github.com/apache/superset/assets/8809047/86dd8c56-0015-4450-ad51-bb49e7f85f2b

After

https://github.com/apache/superset/assets/8809047/88c1f072-1637-47d6-9510-52a4b85556dd



### TESTING INSTRUCTIONS
1. Ensure CACHE_CONFIG is set to use FileSystemCache in config.py/superset_config.py
2. Go to Settings &rarr; Database Connections
3. Edit a DB connection
4. Go to Advanced &rarr; Performance
5. Set values to `SCHEMA CACHE TIMEOUT` and `TABLE CACHE TIMEOUT`
6. Go to SqlLab, select the DB you edited
7. Try to refresh the schema list and table list

### ADDITIONAL INFORMATION


- [x] Has associated issue: Fixes #28735
